src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -151,21 +185,24 @@ Kokkos::View<SizeType *, typename ViewType::device_type> sortObjects(
       n);
   ArborX::iota(space, permute);
 
-#if defined(KOKKOS_ENABLE_CUDA)
-  auto const execution_policy = thrust::cuda::par.on(space.cuda_stream());
-#else
-  auto const execution_policy = thrust::hip::par.on(space.hip_stream());
-#endif
-
-  auto permute_ptr = thrust::device_ptr<SizeType>(permute.data());
-  auto begin_ptr = thrust::device_ptr<ValueType>(view.data());
-  auto end_ptr = thrust::device_ptr<ValueType>(view.data() + n);
-  thrust::sort_by_key(execution_policy, begin_ptr, end_ptr, permute_ptr);
+  thrust::sort_by_key(getThrustExecutionPolicy(space), view.data(),
+                      view.data() + n, permute.data());
```

Comment:
> Using raw pointers to avoid having to spell out device or host?

Yes. Otherwise we would need to specialize traits. And it seems to work fine, so I'm not even sure why we did it in the first place.